src/Inotify.php

```diff
@@ -0,0 +1,261 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Alb\Inotify;
+
+final class Inotify
```

Comment:
If we are going to use a class, we should probably use instance methods as well. I suggest this:

``` php
class Inotify {
    public function __construct(); // inotify_init()
    public function getStream(): resource; // for stream_select() and functional API purposes
    public function addWatch(string $pathname, int $mask): int;
    public function queueLen(): int;
    public function read(): array|false;
    public function rmWatch(int $watchDescriptor): bool;
}
```

The class can be a wrapper over functions, or the other way around. In the latter case, the functions can use stream_get_meta_data() to fetch the Inotify instance from a stream.

Reply:
Have tried to implement this and it doesn't look fun to interact with, as you always have to convert instance of object to resource and back to instance. Overall, it looks overly complicated. It is easier to work directly with streams and simple functions or static functions here without instantiate new objects.

Reply:
Could you give an example where it's required to convert back and forth? I think the resource is only needed when using stream_select().

Reply:
Already deleted all the code)
But it looked somehow like this in functions.

```php
function inotify_init() {
    $inotify = new Inotify();

    return $inotify->getResource();
}

function inotify_add_watch($stream, string $name, int $mask): int {
    $inotify = Inotify::fromStream($stream);

    return $inotify->addWatch($name, $mask);
}
```

>is only needed when using stream_select().

I use Revolt event loop with it and working with Inotify instane directly would also require converting it back and forth
Example:

```php
$inotify = new Inotify();

EventLoop::onReadable($inotify->getResource(), fn($id, $resource) => function () {
    $inotify = Inotify::fromResource($resource);
    $events = $inotify->read();
});
```

Maybe it's not so bad actually, but working with simple streams in async apps just more usual.

Reply:
> it looked somehow like this in functions

This is fine if that's abstracted away in the wrapper functions though

> I use Revolt event loop with it and working with Inotify instane directly would also require converting it back and forth

I see. In this case I think the functions API is better. We don't really need the class if we only have static methods.